query-request.hh

```diff
@@ -232,6 +232,12 @@ public:
     friend std::ostream& operator<<(std::ostream& out, const specific_ranges& ps);
 };
 
+// See docs/design-notes/reverse-reads.md
+void legacy_reverse_slice_to_native_reverse_slice(partition_slice& slice);
+void native_reverse_slice_to_legacy_reverse_slice(partition_slice& slice);
+// Fully reverse slice (forward to native reverse)
+void reverse_slice(partition_slice& slice);
```

Comment:
Wouldn't it be better to use functional-style (pure) functions:
```
partition_slice legacy_reverse...(partition_slice)
```
etc.? I think they are more convenient to use in common scenarios, eg. when you have `const partition_slice&` and want to obtain a reversed slice. And mutability usually tends to make things harder to understand.

This is what I did on my branch:
https://github.com/scylladb/scylla/pull/9281/commits/f78a18ae59c3252adebe7547182a3b993123bb31

Reply:
Yes, I had this version too at one point. Both can work, I don't remember why I switched to the mutable format so it was probably not a good reason. I'll switch to the functional.

Reply:
Another argument is consistency: you're using the functional style in `data_type reversed(data_type)` and in `schema::make_reversed() const`.